Exercise_1.java

```diff
@@ -2,7 +2,35 @@
     // Returns index of x if it is present in arr[l.. r], else return -1 
     int binarySearch(int arr[], int l, int r, int x) 

```

Comment:
Please add time and space complexity for your implementation and review inline comments.